crates/matrix-sdk/src/event_cache/room/mod.rs

```diff
@@ -1190,6 +1194,44 @@ mod private {
             Ok(updates_as_vector_diffs)
         }
 
+        /// If the event has a thread root, ensure the thread root event has a thread summary.
```

Comment:
I think this method would deserve a bit more explanation. What does it do exactly? When is it useful to extract a thread root? What is a thread root? Stuff that could bring some context.